core/src/mast/mod.rs

```diff
@@ -109,7 +95,10 @@ impl MastForest {
 
     /// Adds a decorator to the forest, and returns the associated [`DecoratorId`].
     pub fn add_decorator(&mut self, decorator: Decorator) -> Result<DecoratorId, MastForestError> {
-        self.decorators.push(decorator).map_err(|_| MastForestError::TooManyDecorators)
+        self.debug_info
+            .decorators_mut()
+            .push(decorator)
+            .map_err(|_| MastForestError::TooManyDecorators)
```

Comment:
I would probably add `DebugInfo::add_decorator()` function to make adding decorators an internal concern of the `DebugInfo`.